recipe/meta.yaml

```diff
@@ -1,30 +1,40 @@
 {% set name = "nccl" %}
-{% set version = "2.15.5-1" %}
+{% set version = "2.17.1-1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version|replace("-", ".") }}
 
 source:
   url: https://github.com/NVIDIA/nccl/archive/v{{ version }}.tar.gz
-  sha256: f4ac3c74d469c9cd718f82e1477759785db9b9f8cc9d9ecc103485805b8394a3
+  sha256: 1311a6fd7cd44ad6d4523ba03065ce694605843fd30a5c0f77aa3d911abe706d
+  patches:
+    - remove-trace-call.patch
 
 build:
   number: 0
-  skip: true  # [(not linux) or cuda_compiler_version in (undefined, "None")]
+  skip: true  # [not linux]
   run_exports:
     # xref: https://github.com/NVIDIA/nccl/issues/218
     - {{ pin_subpackage(name, max_pin="x") }}
-  script_env:
-    # for some reason /usr/local/cuda is not added to $PATH in ppc64le's docker image
-    - CUDA_HOME  # [ppc64le or aarch64]
 
 requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
-    - {{ compiler("cuda") }}
+    - {{ compiler("cuda") }}         # [cuda_compiler_version != "12.0"]
+    - cuda-cudart ={{ cuda_compiler_version }}         # [cuda_compiler_version == "12.0"]
+    - cuda-cudart-dev ={{ cuda_compiler_version }}     # [cuda_compiler_version == "12.0"]
+    - cuda-cudart-static ={{ cuda_compiler_version }}  # [cuda_compiler_version == "12.0"]
+    - cuda-nvcc ={{ cuda_compiler_version }}           # [cuda_compiler_version == "12.0"]
```

Comment:
Should these not all be run-deps of `cuda-nvcc` resp. `cuda-nvcc_<target>`? They look like compiler-support libs, I guess it's not the intention that every recipe would have to specify them by hand?

Same for the host-reqs (that are presumably there for the run-exports, which could reasonably be replaced by "strong" run-exports of the cuda-compiler). Compare e.g. the [strong run-exports](https://github.com/conda-forge/nvcc-feedstock/blob/main/recipe/meta.yaml#L40-L47) of the current nvcc metapackage.

PS. It's possible that this PR is only experimenting, but given that a migrator [PR](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4400) now exists, I'm interested in how "real" CUDA 12 PRs would look like currently.

Reply:
Yeah they should (or really they should be dependencies of `cuda-nvcc_{{ target_platform }}`). We've been trying to dogfood the packages to learn about issues and fix them. This is certainly one such issue

The `requirements/host` dependencies here shouldn't be needed either

Yep, as you guessed, this is an experiment, but with the hope of getting something a bit more final (after some iteration on the aforementioned issues)